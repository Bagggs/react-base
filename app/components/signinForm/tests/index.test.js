import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { FormControl } from 'react-bootstrap';
import SigninForm from 'components/signinForm';

describe('Signin form', () => {
  const props = {
    closeModal: jest.fn(),
    signinUser: jest.fn(),
    isLoading: false
  };
  const signinFormComponent = mount(<SigninForm { ...props }/>);

  it('renders correctly', () => {
    const tree = renderer.create(<SigninForm/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('password field', () => {
    const inputPassword = signinFormComponent
      .find(FormControl)
      .filterWhere(n => n.props().name === 'password');

    it('appears error state when password is too short', () => {
      inputPassword.simulate('change', {
        target: { name: 'password', value: 'short' }
      });
      expect(inputPassword.parent().props().validationState).toEqual('error');
    });

    it('appears success state when password is strong', () => {
      inputPassword.simulate('change', {
        target: { name: 'password', value: 'strongPassword' }
      });
      expect(inputPassword.parent().props().validationState).toEqual('success');
    });
  });

  describe('email field', () => {
    const inputEmail = signinFormComponent
      .find(FormControl)
      .filterWhere(n => n.props().name === 'email');

    it('appears error state when email is too short', () => {
      inputEmail.simulate('change', {
        target: { name: 'email', value: 'short' }
      });
      expect(inputEmail.parent().props().validationState).toEqual('error');
    });

    it('appears success state when email is valid', () => {
      inputEmail.simulate('change', {
        target: { name: 'email', value: 'email@example.com' }
      });
      expect(inputEmail.parent().props().validationState).toEqual('success');
    });
  });
});
