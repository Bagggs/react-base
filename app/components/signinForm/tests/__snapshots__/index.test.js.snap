// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin form renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="modal-body"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form-control"
        id="email"
        name="email"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="off"
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
